topics/cluster-spec.md

```diff
@@ -479,6 +479,16 @@ is that:
 * All queries about existing keys are processed by "A".
 * All queries about non-existing keys in A are processed by "B", because "A" will redirect clients to "B".
 
+Starting from Valkey 8.0, the `CLUSTER SETSLOT` command is synchronously replicated to all healthy replicas
+running Valkey version 8.0+. By default, this synchronous replication must complete within 2 seconds.
```

Comment:
It's replicated only if all connected replicas are running valkey 8+, if I remember correctly.

```suggestion
Starting from Valkey 8.0, the `CLUSTER SETSLOT` command is synchronously replicated to all healthy replicas
if all replicas are running Valkey version 8.0+. By default, this synchronous replication must complete within 2 seconds.
```

Reply:
We are not requiring all replica nodes to be running 8.0. We went with eligible (in terms of both version and healthy state) nodes. I think rolling upgrades are a transient state. It should be acceptable to trade reliability for availability during this window, when upgrading from pre-8 to post 8 version.

Reply:
OK, maybe I haven't fully understood this, but we use a shared replication buffer so all replicas receive the same stream of commands, right? In #686, we check for legacy replicas and if one is found, we don't replicate setslot to any replica. Correct me if I'm wrong.

Reply:
I mean, we don't have to highlight this scenario, but the text should still be correct.

Reply:
This is even explained in the PR description of #686:

> Older replicas do not support the CLUSTER SETSLOT command on replicas. If such a replica is found, the replication is skipped and falls back to the old (non-replicated) behavior.

Reply:
Ah, sorry for the confusion. I see that I didn't word it accurately. I can fix that but I think the higher order question is what behavior do we like? "all or none" or "eligible replicas". I am in favor of "eligible replicas" because it is still better than no replication at all. Thoughts?

Reply:
Replicating it only to "eligible replicas" would be better, but that's now what's currently implemented. That would require separate replication streams for legacy and new replicas. We have a single replication stream.

Can we just formulate this in a way that doesn't stress this difference? There's a difference only during rolling upgrade.

Reply:
For example just omit the exact details. How about something like this?

> Starting from Valkey 8.0, the `CLUSTER SETSLOT` command is synchronously replicated if the replicas are running Valkey version 8.0+. By default, this synchronous replication must complete within 2 seconds.

Reply:
sounds good to me